src/Gateway.php

```diff
@@ -69,21 +77,93 @@ public function getDefaultParameters()
      * Create an authorize request.
      *
      * @param array $parameters
-     * @return \Omnipay\Dummy\Message\AuthorizeRequest
+     * @return \Omnipay\Dummy\Message\CreditCardRequest
      */
     public function authorize(array $parameters = array())
     {
-        return $this->createRequest('\Omnipay\Dummy\Message\AuthorizeRequest', $parameters);
+        return $this->createRequest('\Omnipay\Dummy\Message\CreditCardRequest', $parameters);
     }
 
     /**
      * Create a purchase request.
      *
      * @param array $parameters
-     * @return \Omnipay\Dummy\Message\AuthorizeRequest
+     * @return \Omnipay\Dummy\Message\CreditCardRequest
      */
     public function purchase(array $parameters = array())
     {
-        return $this->authorize($parameters);
+        return $this->createRequest('\Omnipay\Dummy\Message\CreditCardRequest', $parameters);
+    }
+
+    /**
+     * @param array $parameters
+     * @return \Omnipay\Common\Message\ResponseInterface
+     */
+    public function completeAuthorize(array $parameters = array())
+    {
+        return $this->createRequest('\Omnipay\Dummy\Message\TransactionReferenceRequest', $parameters);
+    }
+
+    /**
+     * @param array $parameters
+     * @return \Omnipay\Common\Message\ResponseInterface
+     */
+    public function capture(array $parameters = array())
+    {
+        return $this->createRequest('\Omnipay\Dummy\Message\TransactionReferenceRequest', $parameters);
+    }
+
+    /**
+     * @param array $parameters
+     * @return \Omnipay\Common\Message\ResponseInterface
+     */
+    public function completePurchase(array $parameters = array())
+    {
+        return $this->createRequest('\Omnipay\Dummy\Message\TransactionReferenceRequest', $parameters);
+    }
+
+    /**
+     * @param array $parameters
+     * @return \Omnipay\Common\Message\ResponseInterface
```

Comment:
It looks like these docblocks are incorrect - as far as I can tell all the public methods in Gateway always return an AbstractRequest, not a ResponseInterface...